Misc/NEWS.d/next/Core and Builtins/2018-09-20-15-41-58.bpo-34751.Yiv0pV.rst

```diff
@@ -0,0 +1,4 @@
+The hash function for tuples is now based on xxHash.
+This makes a high rate of collisions much less likely.
+Note that the new hash algorithm is different on 32-bit and 64-bit systems.
```

Comment:
Repeating a comment that Github is hiding for some reason - I'm trying to improve the _informativeness_ here:

As you pointed out on BPO, the multiplier was "too small" for 64-bit boxes before. For that reason, the new algorithm (which doesn't have that issue) can be expected to do better on 64-bit boxes in general, quite apart from the handful of very bad special all-platform cases we identified. That's potentially of real interest to users. That the new algorithm is merely somewhat different between its own 32- and 64-bit versions is an implementation detail of no interest to the vast bulk of users.

Reply:
I'll think of a better wording. I just don't want to give the false impression that we mostly improved the hash function on 64-bit systems. We changed the algorithm because of structural weaknesses in the algorithm. The fact that it works slightly better on 64-bit systems is good to have but not nearly as important as the other fix.

Reply:
Well, it's a NEWS blurb, not a paper :wink:  I was happy with my original wording, which put the important point first, then added "and on 64-bit platforms improves tuple hashes in general (the old algorithm was designed for 32-bit hash codes)."  That's informative, which is the point of NEWS.

Reply:
I went with "Additionally, on 64-bit systems it improves tuple hashes in general."

I hope that this is OK.

Reply:
Yup!  Thank you.